replayserver/bookkeeping/bookkeeper.py

```diff
@@ -32,5 +36,35 @@ def build(cls, database, config):
             await self._saver.save_replay(game_id, stream)
             logger.debug(f"Saved replay {game_id}")
             metrics.saved_replays.inc()
+            logger.debug(f"Parsing replay {game_id}")
+            ticks = await self.get_ticks(stream)
+            await self._queries.update_game_stats(game_id, ticks)
         except BookkeepingError as e:
             logger.warning(f"Failed to save replay for game {game_id}: {e}")
+
+    async def get_ticks(self, stream):
+        """Parse the replay and extract the total number of ticks"""
+        loop = asyncio.get_event_loop()
+        return await loop.run_in_executor(
+            None, lambda: _get_replay_ticks(stream.data.bytes()))
+
+
+def _get_replay_ticks(data):
```

Comment:
This should not be in Bookkeeper, but in a separate class passed in as a dependency, so we can unit test properly.

Reply:
Yea I was wondering if I needed to make a whole class for it

Reply:
This method you can remove and call
```
from replay_parser.replay import parse

...

parse(data)
```